src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -94,36 +98,143 @@ private void MoveNext()
             _end = true;
         }
 
-        public void Skip(int length)
+        public void Skip(int byteCount)
         {
-            if (length < 0)
+            if (byteCount < 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
 
-            _consumedBytes += length;
+            _consumedBytes += byteCount;
 
-            while (!_end && length > 0)
+            while (!_end && byteCount > 0)
             {
-                if ((_index + length) < _currentSpan.Length)
+                if ((_index + byteCount) < _currentSpan.Length)
                 {
-                    _index += length;
-                    length = 0;
+                    _index += byteCount;
+                    byteCount = 0;
                     break;
                 }
 
                 var remaining = (_currentSpan.Length - _index);
 
                 _index += remaining;
-                length -= remaining;
+                byteCount -= remaining;
 
                 MoveNext();
             }
 
-            if (length > 0)
+            if (byteCount > 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.length);
             }
         }
+
+        public void SkipTo(Position position)
+        {
+            _currentPosition = position;
+            _nextPosition = position;
+            if (_sequence.TryGet(ref _nextPosition, out ReadOnlyMemory<byte> memory))
+            {
+                _currentSpan = memory.Span;
+            }
+            else
+            {
+                _currentSpan = default;
+            }
+            _index = 0;
+        }
+      
+        public Position? PositionOf(byte value)
```

Comment:
There is same logic in `ROB.Seek`

Reply:
Yeah, but I don't have access to ROB

Reply:
That doesn't mean we have to duplicate everything on reader now.

Reply:
Agreed. We should keep this API minimal.

Reply:
I want to eliminate the old reader. To do this I need a method roughly like the following:
```c#
        public static bool TryReadUntill(ref this BufferReader<ReadOnlyBuffer> reader, out ReadOnlyBuffer bytes, byte delimiter)
        {
            var position = reader.PositionOf(delimiter);
            if (position == null)
            {
                bytes = default;
                return false;
            }
            bytes = new ReadOnlyBuffer(reader.Position, position.Value);
            reader.SkipTo(position.Value);
            reader.Skip(1);
            return true;
        }
```

Even if I change the this parameter to reader over ROB, I am not sure how to implement it. Can you suggest a way to implement it without adding PositionOf to the reader?

Reply:
`while (!reader.IsEnd || reader.Read() != delimiter) {}` until you find `delimiter` and use

Reply:
The method is a Try, but Read has side effects.

Reply:
Copy reader to local and copy back if needed.

Reply:
I think it's ok as a workaround for missing APIs, but it's not a good long term solution. The current implementation is super fast: does not copy large structs, uses vectorized IndexOf to find matches, in common cases where the match is found in the first segment, it simply adjusts the _index (once). Your implementation (workaround) is not even in the same ballpark.

Reply:
I know, I'm not claiming that it is efficient. But we have fast one in ROB and we need to decide how to reuse it and avoid code duplication.